src/sentry/incidents/subscription_processor.py

```diff
@@ -635,6 +635,28 @@ def process_legacy_metric_alerts(
         self.update_alert_rule_stats()
         return fired_incident_triggers
 
+    def has_downgraded(self, dataset: str, organization: Organization) -> bool:
```

Comment:
thanks for pulling this stuff out as a method! reading the contents of the method, it seems like this is filtering out the update based on a few different dimensions.

should this be named something a bit more generic than `has_downgraded`? maybe something like `has_valid_configuration_dataset` and we invert the boolean logic so the if statement can read like `if not self.has_valid_configuration_dataset():`

Reply:
yeah, let's leave the metrics in place. i was mostly just trying to grok the code + the method name, and these weren't 100% locking in for me -- but also don't need to block on any bike shedding for this.